ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -404,7 +404,10 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{ na
     }
 
     private async updateApp(app: appModels.Application) {
-        await services.applications.updateSpec(app.metadata.name, app.spec);
+        const latestApp = await services.applications.get(app.metadata.name);
```

Comment:
what is this doing please?

Reply:
Method load most recent application versions replaces spec and labels and save updated app. This is required to avoid update conflict error (last editor wins)

Reply:
comment?